File: LeetCode/000268.py
from typing import List


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        hashmap = {}
        length = len(nums)
        for num in nums:
            hashmap[num] = True
        for i in range(length + 1):
            if hashmap.get(i) is None:
                return i

File: LeetCode/000268.go
package leetcode

// O(n) | O(n)
func missingNumber(nums []int) int {
	length := len(nums)
	hashmap := make(map[int]bool)
	for _, v := range nums {
		hashmap[v] = true
	}
	result := 0
	for i := 0; i <= length; i++ {
		if _, exists := hashmap[i]; !exists {
			result = i
			break
		}
	}
	return result
}
